code/game/objects/items/weapons/rpd.dm

```diff
@@ -0,0 +1,303 @@
+	list("key1" = "Junction",					"key2" = "6",	"key3" = "5",	"key4" = "4",	"key5" = "junction",		"key6" = "0"),
+	list("key1" = "Heat exchanger",				"key2" = "17",	"key3" = "5",	"key4" = "4",	"key5" = "heunary",			"key6" = "0"))
+
+//Pipename, pipe type, number of unique orientations, name of pipe icon
+var/list/dpipelist = list(
+	list("key1" = "Straight pipe",	"key2" = "0",	"key3" = "2",	"key4" = "conpipe-s"),
+	list("key1" = "Bent pipe",		"key2" = "1",	"key3" = "4",	"key4" = "conpipe-c"),
+	list("key1" = "Junction",		"key2" = "2",	"key3" = "4",	"key4" = "conpipe-j1"),
+	list("key1" = "Y-junction",		"key2" = "4",	"key3" = "4",	"key4" = "conpipe-y"),
+	list("key1" = "Trunk",			"key2" = "5",	"key3" = "4",	"key4" = "conpipe-t"),
+	list("key1" = "Bin",			"key2" = "6",	"key3" = "1",	"key4" = "condisposal"),
+	list("key1" = "Outlet",			"key2" = "7",	"key3" = "4",	"key4" = "outlet"),
+	list("key1" = "Chute",			"key2" = "8",	"key3" = "4",	"key4" = "intake"))
+
+//Procs
+
+/obj/item/weapon/rpd/proc/Activaterpd(soundfile, delay)
+	playsound(loc, soundfile, 50, 1)
+	if(prob(15))
+		spark_system.start()
+	if(delay)
+		cooldown = 1
+		spawn(spawndelay)
+		cooldown = 0
+	return
+
+/obj/item/weapon/rpd/proc/Checkifbent(x)
+	if(x in list(1,3,25,27))
+		return 1
+	else
+		return 0
+
+//This proc puts each necessary icon into the initial user's cache for later use. It then sets initialised = 1 to avoid calling itself unnecessarily.
+/obj/item/weapon/rpd/proc/Initialiseicons()
+	var/F
+	var/I
+	var/list/filenames
+	var/disposalsfile = "icons/obj/pipes/disposal.dmi"
+	var/pipefile = "icons/obj/pipe-item.dmi"
+	for(I in pipelist)
+		if(I["key6"] == "1")//Bendy pipe icons
+			filenames += list(
+			list(pipefile, I["key5"], I["key5"] + "-southeast.png", SOUTHEAST),
+			list(pipefile, I["key5"], I["key5"] + "-southwest.png", SOUTHWEST),
+			list(pipefile, I["key5"], I["key5"] + "-northwest.png", NORTHWEST),
+			list(pipefile, I["key5"], I["key5"] + "-northeast.png", NORTHEAST))
+			continue
+		if(I["key4"] == "2" || I["key4"] == "4") //Dual state icons
+			filenames += list(
+			list(pipefile, I["key5"], I["key5"] + "-north.png", NORTH),
+			list(pipefile, I["key5"], I["key5"] + "-east.png", EAST))
+		if(I["key4"] == "4")//All other pipes that need icons
+			filenames += list(
+			list(pipefile, I["key5"], I["key5"] + "-south.png", SOUTH),
+			list(pipefile, I["key5"], I["key5"] + "-west.png", WEST))
+	for(I in dpipelist)
+		if(I["key3"] == "2" || I["key3"] == "4")
+			filenames += list(
+			list(disposalsfile, I["key4"], I["key4"] + "-north.png", NORTH),
+			list(disposalsfile, I["key4"], I["key4"] + "-east.png", EAST))
+		if(I["key3"] == "4")
+			filenames += list(
+			list(disposalsfile, I["key4"], I["key4"] + "-south.png", SOUTH),
+			list(disposalsfile, I["key4"], I["key4"] + "-west.png", WEST))
+	for(F in filenames)
+		var/icon/iconfile = icon(F[1], icon_state = F[2], dir = F[4])
+		usr << browse_rsc(iconfile, F[3])
```

Comment:
Will the asset cache mean I don't need to goof around with checking icons are initialised? (i.e. will all users automatically get the icons downloaded to their client upon server start?)